Sources/SwiftParser/Patterns.swift

```diff
@@ -354,6 +354,7 @@ extension Parser.Lookahead {
         && !self.at(.keyword(.__owned))
         && !self.at(.keyword(.borrowing))
         && !self.at(.keyword(.consuming))
+        && (!experimentalFeatures.contains(.nonEscapableTypes) || !self.at(.keyword(.resultDependsOn)))
```

Comment:
I think this would be more readable as follows but up to you.

```suggestion
        && !(experimentalFeatures.contains(.nonEscapableTypes) && self.at(.keyword(.resultDependsOn)))
```

Reply:
While not critical, this comment is still open.